src/devices/MAX31856/MAX31856.cs

```diff
@@ -136,14 +133,15 @@ private Temperature ReadTemperature(out byte[] spiOutputData)
         /// <param name="spiOutputData">Spidata read from the device as 16 bytes</param>
         private double ConvertspiOutputDataTempColdJunction(byte[] spiOutputData)
         {
-            var tempRaw = (((spiOutputData[11] & 0x7F) << 8) + spiOutputData[12]);
-            double temperatureColdJunction = tempRaw / 256; // convert decimal ouput to temperature using a constant from the Technical Manual
+            Span<byte> reading = new Span<byte>(spiOutputData, 11, 2);
+            reading[0] = (byte)(spiOutputData[11] & 0x3F);
```

Comment:
I'm not sure what are you doing here... with this you will never get negative number below

Reply:
should this 0x7F?

Reply:
Yes good catch! The first bits on byte 11(0ah)  is for temperature sign. We have to strip it out and combine the remaining 7 bits of byte 11 with byte 12 to get temperature. My mind jumped and I thought it was the first two hence the 0x3f it should be 0x7f